models/marts/core/int_direct_relationships.sql

```diff
@@ -52,11 +53,21 @@ direct_relationships as (
         on all_graph_resources.resource_id = exposures.resource_id
 ),
 
+-- getting the materializationof the parent model to be able to track back chains of views
+direct_relationships_with_materialization as (
```

Comment:
I am wondering if we can avoid using this CTE to map attributes to parents and children, and instead rely on the recursive logic to to this? Feels duplicative to do a self join here as well as in the next downstream model.

Reply:
I had a look and I think that this join will then need to happen anyway in the recursive logic. 
With the current logic, I would believe that having it as a filter at the "import CTE" level allows us to make the recursive DAG for chained_views more performant.

Wouldn't having the logic here make it easier for you for your `ephemeral/view` check for exposures? In that case you don't have to query the fill DAG but just the direct relationships.

Reply:
I agree with Dave here, that it would be better to avoid duplicating this join. If we move this into the recursive logic, we can just `select * from int_all_dag_relationships where distance = 1`. That is the way that we have written the queries for many other fct_ models (_even_ in cases where it's only looking at direct relationships).

Reply:
I agree with Dave here, that it would be better to avoid duplicating this join. If we move this into the recursive logic, we can just `select * from int_all_dag_relationships where distance = 1`. That is the way that we have written the queries for many other fct_ models (_even_ in cases where it's only looking at direct relationships).

Reply:
@b-per and I chatted live about this earlier this week and I _think_ this might be the only way to achieve benoit's intended output -- might be worht having a larger convo next week!